types/node/path.d.ts

```diff
@@ -69,18 +69,19 @@ declare module 'path' {
              * Normalize a string path, reducing '..' and '.' parts.
              * When multiple slashes are found, they're replaced by a single one; when the path contains a trailing slash, it is preserved. On Windows backslashes are used.
              *
-             * @param p string path to normalize.
+             * @param path string path to normalize.
+             * @throws {TypeError} if `path` is not a string.
```

Comment:
On the fence on this, TS itself (sadly) doesn't have checked exception and TS already checks that all segments are strings, this feels redundant in a type checked context.